test/spec_mock_request.rb

```diff
@@ -46,7 +46,7 @@
   end
 
   it "should handle a non-GET request with both :input and :params" do
-    env = Rack::MockRequest.env_for("/", method: :post, input: nil, params: {})
+    env = Rack::MockRequest.env_for("/", method: :post, input: "", params: {})
     env["PATH_INFO"].must_equal "/"
     env.must_be_kind_of Hash
     env['rack.input'].read.must_equal ''
```

Comment:
This line fails, which I think is reasonable, if you specify `nil` input.

Reply:
Seems like the spec should be changed so that when :input is `nil` or not given, `env['rack.input']` is not present. Or there should be a separate spec for that.

Reply:
There are several other specs which I think test the case you are concerned about.